netbox_topology_views/forms.py

```diff
@@ -12,18 +12,19 @@
 from tenancy.models import TenantGroup, Tenant
 from tenancy.forms import TenancyFilterForm
 from django.conf import settings
-from netbox.forms import NetBoxModelFilterSetForm
+from netbox.forms import NetBoxModelFilterSetForm, NetBoxModelForm
 from utilities.forms import (
     TagFilterField,
     DynamicModelMultipleChoiceField,
     MultipleChoiceField,
+    widgets,
 )
+from .models import IndividualOptions, GeneralOptions
 
-allow_coordinates_saving = bool(
-    settings.PLUGINS_CONFIG["netbox_topology_views"]["allow_coordinates_saving"]
+generalOptions, created = GeneralOptions.objects.get_or_create(
```

Comment:
I've had no issues with migrations, even after rolling back and reapplying the migration. It is indeed a problem if you roll back migrations, delete the migration script and use manage.py makemigrations (the workaround is to comment out  problematic lines then). But that should never happen to end users because makemigrations has already been done and the migration script is in the repository and already to be shipped.

Since I'm not familiar with the packaging and installation process, maybe I am missing something.

I don't know actually how to work around this if it is necessary.

Reply:
Yeah, I will have to test some more.
I was also not able to resolve this with a good fix. Will see if I can look at this over the weekend.

Reply:
I know you put al lot of effort into this but i think we should leave the general settings in the config file (for now).
These settings do not change very often and so it is not that important to change them very quickly.
That the individual options have moved out the settings file seems like the most important. 

The  unique_row="general_options" also fe
els a bit strange, but that might just be django.

Reply:
I'd say let's merge just individual options then as it is better than nothing although I've got a different opinion. Problem is that (if I remember correctly) at least one individual option relies on general settings. That must be changed before merging. And of course the docs must be changed.

The general options are a convenient way to change settings without the need of root access. You can use djangos rights management (and Active Directory btw). Furthermore the settings are still present if you transfer the database to another server. In fact, there is no downside as far as I can see. So why not just use it?

About the somewhat strange unique_row: I don't remember what or why it's like it is. I probably can have a look at it next week.

Reply:
I agree on the individual options part, this should be in the interface to be able to quickly change the options.

But the general options are not options that you have to change quickly, to enable the use of the coordinates you need to create the custom field first, so that seems like something you do only once.

The issue i have is that the upgrade does not work from the current latest version to this branch.
Since the only part that seems the block the actual upgrade is the 'general option', i propose we leave this out for now and see if we can fix this in a later version.

Reply:
Fair enough.

Reply:
I completely removed the code for general options and built a new migration script (no. 002). Please try upgrading again.